comit/src/expiries/config.rs

```diff
@@ -0,0 +1,215 @@
+pub enum Protocol {
+    Herc20Hbit,
+    HbitHerc20,
+}
+
+impl Protocol {
+    pub fn config(&self) -> Config {
+        // Since we are targeting Nectar and the App we can set these.
+        let alice = ActorType::Human;
+        let bob = ActorType::Bot;
+
+        match self {
+            Protocol::Herc20Hbit => Config {
+                alice,
+                bob,
+                alpha_required_confirmations: ethereum_required_confirmations(),
+                beta_required_confirmations: bitcoin_required_confirmations(),
+                alpha_block_time: ETHEREUM_BLOCK_TIME_SECS,
+                beta_block_time: BITCOIN_BLOCK_TIME_SECS,
+                alpha_mine_fund_within_n_blocks: ETHEREUM_MINE_WITHIN_N_BLOCKS,
+                beta_mine_fund_within_n_blocks: BITCOIN_MINE_WITHIN_N_BLOCKS,
+                alpha_mine_redeem_within_n_blocks: ETHEREUM_MINE_WITHIN_N_BLOCKS,
+                beta_mine_redeem_within_n_blocks: BITCOIN_MINE_WITHIN_N_BLOCKS,
+            },
+            Protocol::HbitHerc20 => Config {
+                alice,
+                bob,
+                alpha_required_confirmations: bitcoin_required_confirmations(),
+                beta_required_confirmations: ethereum_required_confirmations(),
+                alpha_block_time: BITCOIN_BLOCK_TIME_SECS,
+                beta_block_time: ETHEREUM_BLOCK_TIME_SECS,
+                alpha_mine_fund_within_n_blocks: BITCOIN_MINE_WITHIN_N_BLOCKS,
+                beta_mine_fund_within_n_blocks: ETHEREUM_MINE_WITHIN_N_BLOCKS,
+                alpha_mine_redeem_within_n_blocks: BITCOIN_MINE_WITHIN_N_BLOCKS,
+                beta_mine_redeem_within_n_blocks: ETHEREUM_MINE_WITHIN_N_BLOCKS,
+            },
+        }
+    }
+}
+
+/// Configuration values used during transition period calculations.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct Config {
+    alice: ActorType,
+    bob: ActorType,
+    alpha_required_confirmations: u8,
+    beta_required_confirmations: u8,
+    alpha_block_time: u16,
+    beta_block_time: u16,
+    alpha_mine_fund_within_n_blocks: u8,
+    beta_mine_fund_within_n_blocks: u8,
+    alpha_mine_redeem_within_n_blocks: u8,
+    beta_mine_redeem_within_n_blocks: u8,
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum ActorType {
+    Bot,
+    Human,
+}
+
+impl ActorType {
+    fn period_to_act(&self) -> Duration {
+        match self {
+            ActorType::Bot => 15.minutes(),
+            ActorType::Human => 60.minutes(),
+        }
+    }
+}
+
+impl Config {
+    /// The duration of time it takes for Alice to act.
+    pub fn alice_to_act(&self) -> Duration {
+        self.alice.period_to_act()
+    }
+
+    /// The duration of time it takes for Bob to act.
+    pub fn bob_to_act(&self) -> Duration {
+        self.bob.period_to_act()
+    }
+
+    /// The duration of time it takes for the alpha fund transaction to be
+    /// mined into the blockchain.
+    pub fn mine_alpha_fund_transaction(&self) -> Duration {
```

Comment:
Terminology nit-pick: "mined" to me implies a single confirmation. Might want to change this to "confirm_alpha_fund_transaction".

Reply:
This specifically refers to a single confirmation.

Reply:
I am working on an updated spike version that has the time to act modelled separately for each transaction.

Reply:
But the implementation uses `time_to_mine_n_blocks`? :)

If it is specifically one confirmation, can't you hardcode `n` to `1` then?

Reply:
@tcharding I am not blocking your merge but I would like an answer to this question :)

Reply:
I'm not sure what you are asking @thomaseizinger. `time_to_mine_n_blocks()` calculates a time period that is big enough that we can hope to mine N blocks within this period. It is called from `mine_alpha_fund_transaction` with N set to the same N we should use when calculating the transaction fee. Said differently the transaction fee is calculated to ensure a transaction gets mined within N blocks, we use `time_to_mine_n_blocks` to calculate the time period.

Does this answer your question?

Reply:
No :sweat_smile: 

My point exactly is that this function assumes more than one block is being mined, which is usually referred to as "confirmations".

If the function is called "mine_alpha_fund_transaction" though. Hence, I would assume it gives me the time until the alpha fund transaction is mined which - as I said in my first comment - usually refers to single confirmation aka block but not N blocks which is what this function is actually doing.

Initially it was just a nit but I feel there is a major misunderstanding here:

- "Confirmed" implies to me that N blocks have been mined with N being my confirmation target
- "Mined" implies to me that a single block has been mined that includes this transaction

Hence, for what the function does, I believe it should be called "confirm_alpha_fund_transaction" as it calculates the time until it is confirmed with N blocks and not just included within one.

Reply:
This function does not calculate the time until it is confirmed N times. It calculates the time it would take to mine N blocks because we expect that our transaction will be included _within_ N blocks - therefore we must wait at least N blocks to hope that our transaction is within one of these blocks. Hence it is called 'time to mine'. This is distinct from the time we wait for N confirmations even though both functions rely on the time to mine N blocks.